src/components/DefaultPageLayout.tsx

```diff
@@ -72,13 +90,47 @@ const DefaultPageLayout = ({
     navigate(createHomePageUrl());
   }, [navigate]);
 
+  const handleSave = useCallback(async () => {
+    preSaveDialogDisclosure.onClose();
+    saveProgressDisclosure.onOpen();
+    await saveProjectHex();
+    saveProgressDisclosure.onClose();
+    toast({
+      id: "save-complete",
+      position: "top",
+      duration: 5_000,
+      title: intl.formatMessage({ id: "saving-toast-title" }),
+      status: "info",
+    });
+  }, [
+    preSaveDialogDisclosure,
+    saveProgressDisclosure,
+    saveProjectHex,
+    toast,
+    intl,
+  ]);
+
+  const handleSaveClick = useCallback(() => {
+    if (settings.showPreSaveHelp) {
+      preSaveDialogDisclosure.onOpen();
+    } else {
+      void handleSave();
+    }
+  }, [handleSave, preSaveDialogDisclosure, settings.showPreSaveHelp]);
+
   return (
     <>
       {/* Suppress connection and train dialogs when MakeCode editor is open */}
       {!isEditorOpen && (
         <>
           <ConnectionDialogs />
           <TrainModelDialogs />
+          <SaveHelpDialog
```

Comment:
This will conflict with my PR but hopefully not too badly as it's conceptually similar.

Reply:
Yes, I also noticed that 😄